File: backend/src/models/Customer.js
const mongoose = require('mongoose');

const customerSchema = new mongoose.Schema(
  {
    name: { type: String, required: true, trim: true },
    phone: { type: String, required: true, trim: true },
    address: { type: String, required: true, trim: true },
    pendingAmount: { type: Number, default: 0, min: 0 },
    active: { type: Boolean, default: true },
    ownerId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  },
  { timestamps: true }
);

customerSchema.index({ ownerId: 1, phone: 1 }, { unique: true });
customerSchema.index({ ownerId: 1, active: 1, name: 1 });

module.exports = mongoose.model('Customer', customerSchema);

File: frontend/lib/models/customer.dart
class CustomerModel {
  final String id;
  final String name;
  final String phone;
  final String address;
  final double pendingAmount;
  final bool active;

  CustomerModel({
    required this.id,
    required this.name,
    required this.phone,
    required this.address,
    required this.pendingAmount,
    required this.active,
  });

  factory CustomerModel.fromJson(Map<String, dynamic> json) {
    return CustomerModel(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
      address: json['address'] ?? '',
      pendingAmount: (json['pendingAmount'] ?? 0).toDouble(),
      active: json['active'] ?? true,
    );
  }

  Map<String, dynamic> toJson() => {
        'name': name,
        'phone': phone,
        'address': address,
        'pendingAmount': pendingAmount,
        'active': active,
      };
}
